tests/regression/test_assemble.py

```diff
@@ -291,3 +291,28 @@ def test_assemble_vector_rspace_one_form(mesh):
     U = inner(u, u)*dx
     L = derivative(U, u)
     assemble(L)
+
+
+def test_assemble_sparsity():
+    mesh = UnitSquareMesh(2, 2, quadrilateral=True)
+    V = FunctionSpace(mesh, "CG", 1)
+    W = V * V * V
+    u = TrialFunction(W)
+    v = TestFunction(W)
+    A = assemble(inner(u, v) * dx, mat_type="nest")
+    for i in range(len(W)):
+        for j in range(len(W)):
+            if i != j:
+                assert np.all(A.M.sparsity[i][j].nnz == np.zeros(9, dtype=IntType))
+
+
+def test_assemble_sparsity_with_bc():
```

Comment:
Do we need an `EquationBC` test too?

Reply:
Correctness of the changes introduced in this PR is checked by the existing tests (including the ones for `EquationBC`) as `Sparsity` is used in most places.